iterators.tex

```diff
@@ -276,7 +276,7 @@
   template <class I>
   concept bool Readable@\newtxt{() \{}\oldtxt{ =}@
     @\newtxt{return}@ Semiregular<I>@\newtxt{()}@ &&
-      requires (I i) {
+      requires (@\newtxt{const }@I i) {
```

Comment:
Does "semantically equivalent" mean they must not have any observable side-effects that are different? If `*declval<T&>()`, `*declval<const T&>()`, `*declval<T&&>()`, and/or `*declval<const T&&>()` return different types, are all non-modifying, and return something convertible to `const ValueType<I>&`, can those conversion operators write different things to `cout`, or would `T` no longer model `Readable`?

Do we define "semantic equivalence" anywhere? Side-effects are allowed by equality-preserving expressions. They should probably be allowed here.

Perhaps we can tie the definition of "semantic equivalence" to "equality preserving".